pkg/rancher-desktop/assets/scripts/moproxy.initd

```diff
@@ -19,6 +19,8 @@ description_reload="Reload the proxy list."
 : ${moproxy_args:=${MOPROXY_ARGS:-""}}
 # Comma-seperated list of port traffic to redirect to moproxy
 : ${ports_redirected:=${MOPROXY_REDIRECTED_PORT:-"80,443"}}
+# Comma-seperated list of hostname to not redirect to the proxy
+: ${noproxy_rules:=${MOPROXY_NOPROXY:-"localhost,172.17.0.0/16"}}
```

Comment:
@tperale thanks for making the changes, however, I see an issue with this approach. This PR aims to redirect anything destined for the docker container's mapped port on localhost to the actual docker container listening port on that subnet (`172.17.0.0/16`). While this will work just for docker, it will not work for other things. E.g. containerd backend (since it will have a different subnet). Furthermore, we also have inter-process APIs that talk to each other using localhost. e.g [guest agent](https://github.com/rancher-sandbox/rancher-desktop-agent/blob/4616f600c9a58785ef52933b5e975086ccb258a8/pkg/tracker/apitracker.go#L74) calls `http://192.168.127.1:80/services/forwarder/expose` on [rancher-desktop-networking](https://github.com/rancher-sandbox/rancher-desktop-networking/blob/c8905b069011074cde795e4bb81c1c4067c421d3/cmd/host/switch_windows.go#L141) to expose container ports. Overall, the exclusion fix to for localhost to bypass proxy would need to be more robust. Thanks.

Reply:
@Nino-K Can we exclude every address mentioned here https://en.wikipedia.org/wiki/Reserved_IP_addresses#Reserved_IPv4_addresses ?

Reply:
Maybe, that would work. But please let me think about this to make sure there are no consequences to doing this.

Reply:
Of course 👍